src/common/addr_parsing.c

```diff
@@ -42,8 +42,8 @@ int safe_cat(char **pstr, int *plen, int pos, const char *str2)
     //printf("safe_cat '%s' max %d pos %d '%s' len %d\n", *pstr, *plen, pos, str2, len2);
   }
```

Comment:
:fearful: 

Why is this function doing a while loop of reallocs when a single one would do?? I think we can simplify this whole function to:

````
#define BUFFER_ROUND_UP(u) (((u) + (128-1)) & (~(128-1)))
size_t safe_cat(char** pstr, size_t pos, const char* str2)
{
  size_t str2len = strlen(str2);
  *pstr = realloc(*pstr, BUFFER_ROUND_UP(pos+str2len+1));
  ceph_assert(*pstr);
  memcpy(*pstr+pos, str2, str2len+1);
  return pos+str2len;
}
````

(untested of course and with better typing. also `plen` is completely unnecessary)

Reply:
@batrick i 

- want to limit the scope of this change. so i cannot change the function signature.
- want to keep the existing behavior. so i cannot to use realloc if the size of `*ptr` is large enough
- want to use C language. so i cannot use `ceph_assert()`.

Reply:
i updated the implementation a little bit to avoid the loop. but personally, it's far from what i expected. as this does not help with readability. and i checked the callers of this function, this function is not in the critical path. anyway...

Reply:
Okay. Just one nit:

>want to keep the existing behavior. so i cannot to use realloc if the size of *ptr is large enough

The realloc in my above implementation will be a no-op if the size does not change.

Anyway,  your change looks good.

Reply:
> Okay. Just one nit:
> 
> > want to keep the existing behavior. so i cannot to use realloc if the size of *ptr is large enough
> 
> The realloc in my above implementation will be a no-op if the size does not change.
> 
> Anyway, your change looks good.

yes, i want to save an unnecessary `realloc()` call which does not exist in previous implementation.